### ISSUE TITLE
6.16.0 has introduced build failure with "export default async function"

### ISSUE THREAD

**Issue by MartinDoyleUK** [babel/babylon#374]
### Input Code

```js
export default async function myFunc() {}
```

### Babylon/Babel Configuration

```js
{
  "presets": [
    ["env", {
      "targets": {
        "node": 6.6
      }
    }]
  ],
  "sourceMaps": "inline"
}
```

### Expected Behavior

I expect the file to build without error.

### Current Behavior

Without modifying the file, I get an error when building of `Property declaration of ExportDefaultDeclaration expected node to be of a type ["FunctionDeclaration","ClassDeclaration","Expression"] but instead got "VariableDeclaration"`.

If I wrap the export with brackets, e.g. `export default (async function myFunc() {});` then it will build successfully.

### Your Environment

| software         | version
| ---------------- | -------
| Babylon          |  6.16.0
| node             | 6.9.4
| npm              | 4.0.5
| Operating System | macOS 10.12

**Comment by MartinDoyleUK** [comment]
Just to clarify, if I install babylon@6.15.0 in my project, the build succeeds.

**Comment by hzoo** [comment]
Releasing v6.16.1 until we actually get the time to figure this out.

**Comment by matAtWork** [comment]
Whilst investigating https://github.com/MatAtBread/fast-async/issues/33, I ended up with this simple test:

(empty .babelrc - no transforms at all)
```
$ echo 'export default async function bar() {}' | ./node_modules/babel-cli/bin/babel.js
export default (async function bar() {});

$ echo 'export default function bar() {}' | ./node_modules/babel-cli/bin/babel.js
export default function bar() {}
```

It appears babylon 6.16.1 incorrectly parses the `async function` as a `FunctionExpression`, not a `FunctionDeclaration`

**Comment by danez** [comment]
@matAtWork This is exactly what we tried to fix in 6.16.0 but had to revert in 6.16.1, as i was causing regressions in babel. This will be fixed with #377

**Comment by danez** [comment]
This is going to be fixed in 7.0
